### ISSUE TITLE
ganeti checks existence of lvm volume group on nodes with vm_capable=False

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#488]
Originally reported of Google Code with ID 432.

```
This is debian bug 650664 

Package: ganeti2
Version: 2.4.2-1~bpo60+1
Severity: normal

While the Ganeti administrators guide says that the unsetting of the vm_capable flag 
should deactivate vm checks, the check for the LVM volume group is still executed 
on this node if I add LVM storage to my cluster. The addition of a vg_name fails.

root@chur:~# gnt-cluster version
Software version: 2.4.2
Internode protocol: 2040000
Configuration format: 2040000
OS api version: 20
Export interface: 0

root@chur:~# gnt-cluster --version
gnt-cluster (ganeti v2.4.2) 2.4.2

root@chur:~# gnt-cluster info
Cluster name: ganeti.example.com
Cluster UUID: f03ca2cf-28ba-4eff-8c9e-a7884467a247
Creation time: 2011-11-30 16:08:46
Modification time: 2011-12-01 18:51:51
Master node: chur
Architecture (this node): 64bit (x86_64)
Tags: (none) 
Default hypervisor: kvm
Enabled hypervisors: kvm
Hypervisor parameters:
  - kvm:
      acpi: True
      boot_order: disk
      cdrom2_image_path:
      cdrom_disk_type:
      cdrom_image_path:
      disk_cache: default
      disk_type: paravirtual
      floppy_image_path:
      initrd_path:
      kernel_args: ro
      kernel_path:
      kvm_flag:
      mem_path:
      migration_bandwidth: 32
      migration_downtime: 30
      migration_mode: live
      migration_port: 8102
      nic_type: paravirtual
      root_path: /dev/vda1
      security_domain:
      security_model: none
      serial_console: True
      usb_mouse:
      use_chroot: False
      use_localtime: False
      vhost_net: False
      vnc_bind_address:
      vnc_password_file:
      vnc_tls: False
      vnc_x509_path:
      vnc_x509_verify: False
OS-specific hypervisor parameters:
OS parameters:
Hidden OSes:
Blacklisted OSes:
Cluster parameters:
  - candidate pool size: 10
  - master netdev: gan-br0
  - lvm volume group: None
  - lvm reserved volumes: (none)
  - drbd usermode helper: None
  - file storage path: /srv/ganeti/file-storage
  - maintenance of node health: False
  - uid pool:
  - default instance allocator:
  - primary ip version: 4
  - preallocation wipe disks: False
Default node parameters:
    oob_program: None
Default instance parameters:
  - default:
      auto_balance: True
      memory: 128
      vcpus: 1
Default nic parameters:
  - default:
      link: xen-br0
      mode: bridged

root@chur:~# gnt-node info chur
Node name: chur
  primary ip: 10.42.0.156
  secondary ip: 10.42.0.156
  master candidate: True
  drained: False
  offline: False
  master_capable: True
  vm_capable: False
  node parameters:
    - oob_program: default (None)

root@chur:~# gnt-cluster modify -d --vg-name=ganeti
2011-12-01 18:57:33,966: gnt-cluster modify pid=2887 cli:1927 INFO run with arguments '-d --vg-name=ganeti'
Thu Dec  1 18:57:34 2011  - WARNING: Error while gathering data on node altels-1.int.cirrax.com (ignoring node): Node is marked offline
2011-12-01 18:57:34,803: gnt-cluster modify pid=2887 cli:1936 ERROR Error during command processing
Traceback (most recent call last):
  File "/usr/lib/pymodules/python2.6/ganeti/cli.py", line 1932, in GenericMain
    result = func(options, args)
  File "/usr/lib/pymodules/python2.6/ganeti/client/gnt_cluster.py", line 810, in SetClusterParams
    SubmitOpCode(op, opts=opts)
  File "/usr/lib/pymodules/python2.6/ganeti/cli.py", line 1727, in SubmitOpCode
    reporter=reporter)
  File "/usr/lib/pymodules/python2.6/ganeti/cli.py", line 1708, in PollJob
    return GenericPollJob(job_id, _LuxiJobPollCb(cl), reporter)
  File "/usr/lib/pymodules/python2.6/ganeti/cli.py", line 1530, in GenericPollJob
    errors.MaybeRaise(msg)
  File "/usr/lib/pymodules/python2.6/ganeti/errors.py", line 448, in MaybeRaise
    raise errcls, args
OpPrereqError: (u"Error on node 'chur': volume group 'ganeti' missing", u'environment_error')
Failure: prerequisites not met for this operation:
error type: environment_error, error details:
Error on node 'chur': volume group 'ganeti' missing

```
Originally added on 2013-04-22 12:15:36 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-04-26 08:29:14 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-05-01 12:54:25 +0000 UTC.

Added Labels: Priority-Medium
Removed Labels: Priority-Low

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-05-02 12:49:48 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 7d60c3b508d7cad9053443adbdfcfb91191e5e9f
Author: Helga Velroyen <helgav@google.com>
Date:   Thu May 2 15:21:27 2013 +0200

    gnt-cluster modify: check vg name only on vm-capable nodes
    
    This fixes issue 432: when setting the volume group name, the presence
    of the volume group was checked on all nodes. This patch now restricts
    the check only to the vm-capable ones.
    
    Signed-off-by: Helga Velroyen <helgav@google.com>
    Reviewed-by: Guido Trotter <ultrotter@google.com>

```
Originally added on 2013-05-03 09:06:42 +0000 UTC.

Mentioned Issue with GoogleCode ID 432 has been migrated to GitHub with ID #488.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:49 +0000 UTC.

Changed State: Released
